core/controllers/base.py

```diff
@@ -258,14 +279,11 @@ def dispatch(self):
                 b'https://oppiatestserver.appspot.com', permanent=True)
             return
 
-        if (feconf.ENABLE_MAINTENANCE_MODE and
-                not self.current_user_is_super_admin):
+        if (feconf.ENABLE_MAINTENANCE_MODE
+                and not self.current_user_is_super_admin
+                and self.request.path not in AUTH_HANDLER_PATHS):
```

Comment:
So just to confirm: the new experience is that all users will be able to access the auth handlers in maintenance mode but the redirection should kick in and prevent them from proceeding unless they're a super admin?

If that's our only option, then I guess we need to do it. We'll just need to be careful whenever we're changing the login/session flow itself since bringing the site down for maintenance won't stop traffic to those endpoints.

Reply:
Yep that's right, and agreed. I think it's a necessary evil since we can't rely on the login page being non-Angular anymore.